packages/mobile/src/identity/reducer.ts

```diff
@@ -202,9 +209,21 @@ export const reducer = (
         },
       }
     case Actions.UPDATE_E164_PHONE_NUMBER_SALT:
+      const newIdentifierToE164Numbers: IdentifierToE164NumberType = {}
+      for (const e164Number of Object.keys(action.e164NumberToSalt)) {
+        const pepper = action.e164NumberToSalt[e164Number]
+        if (pepper) {
+          const phoneHash = getPhoneHash(e164Number, pepper)
+          newIdentifierToE164Numbers[phoneHash] = e164Number
+        }
+      }
```

Comment:
Instead of making a new property in the redux store, can we instead abstract this behind the `identifierToe164NumberSelector` using `createSelector` from `reselect`?

Seems like that will be possible given this can be entirely derived from what we already have in state and will save us from increasing the state side + creating a new action

Reply:
It would be possible, but we would have to parse the entire `e164NumberToSalt` on the reducer and call `getPhoneHash` for each entry. I guess this is a space vs time trade-off. The size of this mapping shouldn't be too big, it would consist of the contacts you have at most.

We're also not adding a new action, we're reusing the same action that adds to the `e164NumberToSalt` mapping.

Wtyt?

Reply:
Yeah I see your point.

`e164NumberToSalt` is an object so lookups are in constant time + `reselect` memoizes so this probably nets out to be roughly the same either way. I guess it wouldn't really simplify the code as much as I thought it might so probably not worth the effort to make the change

Reply:
I thought over this a bit more and I was convinced of your approach. You're right that it's memoized so it shouldn't be a big hit and there's probably not going to be a huge number of items anyways.

It does reduce the amount of code and avoids the migration too so I think this is better, made the change :)